samza-core/src/test/java/org/apache/samza/container/TestContainerHeartbeatMonitor.java

```diff
@@ -93,6 +103,79 @@ public void testDoesNotCallbackWhenHeartbeatAlive() throws InterruptedException
     verify(this.scheduler).shutdown();
   }
 
+  @Test
+  public void testReestablishConnectionWithNewAM() throws InterruptedException {
+    String containerExecutionId = "0";
+    String newCoordinatorUrl = "http://some-host-2.prod.linkedin.com";
+    this.containerHeartbeatMonitor =
+        spy(new ContainerHeartbeatMonitor(this.onExpired, this.containerHeartbeatClient, this.scheduler, COORDINATOR_URL,
+            containerExecutionId, coordinatorStreamStore, true, 5, 10));
+    CoordinatorStreamValueSerde serde = new CoordinatorStreamValueSerde(SetConfig.TYPE);
+    ContainerHeartbeatResponse response1 = new ContainerHeartbeatResponse(false);
+    ContainerHeartbeatResponse response2 = new ContainerHeartbeatResponse(true);
```

Comment:
Can we extract these to class level constants?
Perhaps, name them `FAILURE_RESPONSE` and `SUCCESS_RESPONSE` to be explicit.